pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -242,25 +244,65 @@ func (c *Controller) updateServiceNodePortAddresses(svcs ...*model.Service) bool
 		return false
 	}
 	for _, svc := range svcs {
+		c.RLock()
+		_, found := c.nodeSelectorsForServices[svc.Hostname]
+		isNodePortGateway := found && c.servicesMap[svc.Hostname] != nil
```

Comment:
why not reuse `isNodePortGatewayService`

Reply:
`isNodePortGatewayService` takes a Kubernetes service as parameter. So this had to be done.